### ISSUE TITLE
No 'Access-Control-Allow-Origin' header is present

### ISSUE THREAD

**Issue by haider288** [Rohithzr/payfort-node#2]
just followed the code sample provided, yet i'm getting the below error when testing my app:

Fetch API cannot load https://sbcheckout.payfort.com/FortAPI/paymentPage. Redirect from 'https://sbcheckout.payfort.com/FortAPI/paymentPage' to 'https://sbcheckout.payfort.com/FortAPI/paymentPageC?S=1&token=BJAWM6LI842ABBN1UCSRTOP80BS4JA&' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://***.com' is therefore not allowed access. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

TIA

**Comment by rohithzr** [comment]
are you trying from locahost or server?

**Comment by haider288** [comment]
my app is hosted on heroku server, HTTPS
if it helps you here is a reply from payfort support team, i opened a ticket with them:

> If the request is coming like redirection https post , then you will not get the CORS policy warning . 
> we have lot of merchants integrating without this error since the request is coming as https post . 
> The Ajax for example is not allowed due to security reasons and you will get the same error .

**Comment by rohithzr** [comment]
have you added your authorised url in ur payfort admin panel? i think that is important too

**Comment by haider288** [comment]
i just tried it, same result

**Comment by rohithzr** [comment]
show the code that you are trying to execute. the whole flow that you are following.

**Comment by haider288** [comment]
one page:

`import React , {Component} from 'react';
import {Card, CardSection, Input, Button, Spinner} from './common';

var payfort = require("payfort-node");

class Payfort extends Component{

        // create client


        onButtonPress(){

          var client = payfort.create_client("development", {
            access_code : "xxxxxxxxxxxxxxxx",
            merchant_identifier : "xxxxxxxxxxxxxxx",
            passphrase : "xxxxxxxxxxxxxx"
            //purchase_url : "send this only to override default urls"
          });

          var purchaseData = {
            "amount": "22",
            "command" : "PURCHASE", // PURCHASE OR AUTHORIZATION
            "currency": "AED",
            "customer_email": "dd@hotmail.com",
            "customer_name": "Haider",
            "language": "en",
            "return_url": "http://localhost:3000/lib",
            "merchant_reference": "21221"
          };

          console.log("going to send");
          //call payfort API
          payfort.send_request(client, purchaseData, function(err, response){
            console.log("response");
            console.log(response);
            if(err){
                console.log("got an error");
                console.log(err);
              }
              console.log("outside errors");
          });
        }

        render(){
          return (
              <Card>
                <CardSection>
                    <Button onPress={this.onButtonPress.bind(this)}>
                      Payfort
                    </Button>
                </CardSection>
              </Card>
          );
        }
}

export default Payfort;
`

**Comment by rohithzr** [comment]
give me a little time to test it? can you provide me (mail me) a test key that i can use, that will be helpful

**Comment by haider288** [comment]
take ur time, i've sent you an email to ur live.com account

**Comment by rohithzr** [comment]
@haider288 i did not recieve an email, can you please verify? send one to rohit@trumpt.in

**Comment by haider288** [comment]
email sent

**Comment by haider288** [comment]
hi, did you have time to look at the above issue?

**Comment by rohithzr** [comment]
@haider288 i am testing it now.

**Comment by rohithzr** [comment]
@haider288 are you able to get the checkout url from the the code? i am able to get that

**Comment by haider288** [comment]
Yes, but when I get redirected to it I get the error mentioned previously 

Regards,
Haidar

**Comment by rohithzr** [comment]
@haider288 the process is
you run `payfort.send_request` and get a url in response and manually redirect client to there

**Comment by haider288** [comment]
I don't know what you mean by "manually" redirect the user, I just followed the sample code you posted. 
Other than the code is there anything else I should know?? Something you didn't document ??

Regards,
Haidar

**Comment by rohithzr** [comment]
@haider288 payfort has changed the way their API works now. you need to create a form and call the url from client side. 

I am looking into this issue and will update or remove repo accordingly.

**Comment by hicham-alsayed-ahmad** [comment]
hello, any updates ? 
is there anyway to use direct ionic  http function?

**Comment by rohithzr** [comment]
Not maintained anymore, feel free to contribute
